pkg/pagerduty/types.go

```diff
@@ -0,0 +1,15 @@
+package pagerduty
+
+// Incident represents a PagerDuty Incident
+type Incident struct {
```

Comment:
Do we really want to redefine the pagerduty-go types? For me it doesn't make any sense.
We can just pick it up from here: https://github.com/PagerDuty/go-pagerduty/blob/433447fbcc6577b0faec04b3c613d04e51763ad4/incident.go#L75
As we will need to maintain the duplicate definition.
Wrapping the communication to Pagerduty makes sense to me, as we are going to mock the functions for testing. But that doesn't apply to the data types or? 
However, if I am thinking about it now even wrapping the pagerduty-go library doesn't make too much sense IMO. Just defining an interface of the PD SDK in our code would simply also do its job and we would be able to mock and catch any call that is done to the library during runtime.

Reply:
I think that working this way ensures that ANY change we want to make, if it's in the datastructure or the api calls can happen without changing the code outside this function

Reply:
got it: For the case, we update the SDK we can simply change our wrapper function. However, not sure whether it is worth it for the data structures. In gcp-project-operator, we also simply used the SDK types.

Reply:
I know gcp-project-operator doesn't do that. but by doing it my way the code outside the pagerduty package (even the tests) doesn't use the sdk types (aside of creating the client)